fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogStore.kt

```diff
@@ -0,0 +1,284 @@
+import mozilla.components.concept.engine.translate.Language
+import mozilla.components.concept.engine.translate.TranslationError
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * The [TranslationsDialogStore] holds the [TranslationsDialogState] (state tree).
+ *
+ * The only way to change the [TranslationsDialogState] inside
+ * [TranslationsDialogStore] is to dispatch an [Action] on it.
+ */
+class TranslationsDialogStore(
+    initialState: TranslationsDialogState,
+    middlewares: List<Middleware<TranslationsDialogState, TranslationsDialogAction>> = emptyList(),
+) : Store<TranslationsDialogState, TranslationsDialogAction>(
+    initialState,
+    TranslationsDialogReducer::reduce,
+    middlewares,
+) {
+    init {
+        dispatch(TranslationsDialogAction.FetchSupportedLanguages)
+    }
+}
+
+/**
+ * The current state of the Translations bottom sheet dialog.
+ *
+ * @property isTranslated The page is currently translated.
+ * @property isInProgress The page is currently attempting a translation.
+ * @property positiveButtonType Can be enabled,disabled or in progress.
+ * @property error An error that can occur during the translation process.
+ * @property dismissDialogState Whether the dialog bottom sheet should be dismissed.
+ * @property initialFrom Initial "from" language, based on the translation state and page state.
+ * @property initialTo Initial "to" language, based on the translation state and page state.
+ * @property fromLanguages Translation menu items to be shown in the translate from dropdown.
+ * @property toLanguages Translation menu items to be shown in the translate to dropdown.
+ * @property translatedPageTitle Title of the bottom sheet dialogue if the page was translated.
+ */
+data class TranslationsDialogState(
+    var isTranslated: Boolean = false,
+    val isInProgress: Boolean = false,
+    val positiveButtonType: PositiveButtonType? = null,
+    val error: TranslationError? = null,
+    val dismissDialogState: DismissDialogState? = null,
+    val initialFrom: Language? = null,
+    val initialTo: Language? = null,
+    val fromLanguages: List<Language>? = null,
+    val toLanguages: List<Language>? = null,
+    val translatedPageTitle: String? = null,
+) : State
+
+/**
+ * Action to dispatch through the `TranslationsStore` to modify `TranslationsDialogState` through the reducer.
+ */
+sealed class TranslationsDialogAction : Action {
+    /**
+     * When FetchSupportedLanguages is dispatched, an [TranslationsAction.OperationRequestedAction]
+     * will be dispatched to the [BrowserStore]
+     */
+    object FetchSupportedLanguages : TranslationsDialogAction()
+
+    /**
+     * Invoked when the [TranslationsDialogStore] is added to the fragment.
+     */
+    object InitTranslationsDialog : TranslationsDialogAction()
```

Comment:
nit: Recommend ordering the `InitTranslationsDialog` to be the first item in this list of Actions.

Reply:
Done